packages/template-retail-react-app/app/components/island/README.md

```diff
@@ -0,0 +1,50 @@
+# 🏝️️Islands
+
+Inspired by [Astro’s islands architecture](https://docs.astro.build/en/concepts/islands/), this component is intended to give developers explicit and fine-granular control over the hydration behavior of their experiences.
+
+> ℹ️ Hydration is the process of making a server-rendered HTML page interactive, i.e. bridging the gap between server-side rendering (SSR) and client-side interactivity.
+
+# 😶‍🌫️ Hydration’s Curse
+
+Our Composable Storefront applications are server-rendered Single-Page Applications (SPAs). During hydration, the client takes over the server-rendered HTML, restores client-side interactivity and transitions the application into its fully-functional SPA mode.
+
+Behind the scenes, this **restoration of interactivity is computationally very expensive**. This may sound surprising, since hydration from the outside seems to only be about applying certain event listeners to certain elements. Unfortunately, the reality is much more complex. Even though the HTML/DOM already exists, **React must reconcile and often re-execute render logic across the entire application**, which can feel like a re-render and be similarly costly.
+
+In this way, discrepancies between the HTML generated on server and client are detected, reported, and corrected/replaced. On the flipside, the consequence of this behavior is that large amounts of server-rendered HTML directly correlate with extensive re-rendering processes on the client. The results are long-running JavaScript processes blocking the main thread, thereby preventing other (bootstrapping) operations or user interactions from occurring in a timely manner.
+
+> **💩 Gotcha:** With the good intention of optimizing important (Core) Web Vitals of your website, such as [FCP](https://web.dev/articles/fcp) or [LCP](https://web.dev/articles/lcp), you immediately run into problems with other performance metrics such as [TTI](https://web.dev/articles/tti) or [TBT](https://web.dev/articles/tbt).
+
+# 🚱 Partial Hydration
+
+Hydration in React by default is an all-or-nothing game. But if this game is about lost customers and revenue, it’s an expensive game in multiple ways.
```

Comment:
Does this mean that wrapping a portion of your project's code in an `<Island />` is capable of yielding out-of-sync React behaviors when it comes to statefulness? If "yes" we should probably warn the reader at this point